.github/actions/setup-job-docker/action.yml

```diff
@@ -20,6 +20,14 @@ runs:
       with:
         working-directory: ${{ env.RUST_WORKSPACE }}
 
+    - name: Configure cargo
+      shell: bash
+      run: |
+        # We do this since multiple jobs are run on the same CI build server,
```

Comment:
❔would it be possible to limit the cpu resources a single docker container can use instead of limit the cargo jobs?

Reply:
I think that this would be possible in theory (with docker) but not in practice (with the github runners spawning the docker containers). It would certainly be the cleaner solution, though, so we could research this as a follow-up. 📌 maybe?

Reply:
> I think that this would be possible in theory (with docker) but not in practice (with the github runners spawning the docker containers).

Ah true that could be a problem. As far as I remember, the settings that you can set on the github runner are very limited

Reply:
github allows to set some options for the container including cpu. We might also try something with cgroups. but it could be an optimization of this.

How 6 was choosen?

Reply:
Fair question. Currently it's approx. `$number_of_cores / $number_of_runners` :) I definitely think this _could_ be something we might want to look into, but only if just setting a fixed number doesn't work well.